WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,273 @@
+/// either *cost* or *count* limit is reached. The sweeps are performed periodically.
+///
+/// - important: It's possible to have more than one instance of ``DiskCache`` with
+/// the same path but it is not recommended.
+public actor DiskCache {
+    /// The path for the directory managed by the cache.
+    public nonisolated let rootURL: URL
+
+    /// The cache configuration.
+    public nonisolated let configuration: Configuration
+
+    public struct Configuration {
+        /// Size limit in bytes. `100 Mb` by default.
+        ///
+        /// Changes to the size limit will take effect when the next LRU sweep is run.
+        public var sizeLimit: Int = 1024 * 1024 * 100
+
+        /// When performing a sweep, the cache will remote entries until the size of
+        /// the remaining items is lower than or equal to `sizeLimit * trimRatio` and
+        /// the total count is lower than or equal to `countLimit * trimRatio`. `0.7`
+        /// by default.
+        var trimRatio = 0.7
+
+        /// The number of seconds between each LRU sweep. 30 by default.
+        /// The first sweep is performed right after the cache is initialized.
+        ///
+        /// Sweeps are performed in a background and can be performed in parallel
+        /// with reading.
+        public var sweepInterval: TimeInterval = 30
+
+        /// The delay after which the initial sweep is performed. 10 by default.
+        /// The initial sweep is performed after a delay to avoid competing with
+        /// other subsystems for the resources.
+        var initialSweepDelay: TimeInterval = 10
+
+        /// Initializes the configuration.
+        ///
+        /// - parameter sizeLimit: Size limit in bytes. `100 Mb` by default.
+        public init(sizeLimit: Int = 1024 * 1024 * 100) {
+            self.sizeLimit = sizeLimit
+        }
+    }
+
+    /// Creates a cache instance with a given `name`. The cache creates a directory
+    /// with the given `name` in a `.cachesDirectory` in `.userDomainMask`.
+    public init(name: String, configuration: Configuration = .init()) {
+        let cachesURL = DiskCache.getCachesURL()
+        self.init(url: cachesURL.appendingPathComponent(name, isDirectory: true), configuration: configuration)
+    }
+
+    private static func getCachesURL() -> URL {
+        guard let cachesURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            DDLogError("Failed to instantiate cache. Caches directory not available")
+            return URL(fileURLWithPath: "/dev/null") // This should never happen
+        }
+        return cachesURL
+    }
+
+    /// Creates a cache instance with a given root URL.
+    public init(url: URL, configuration: Configuration = .init()) {
+        self.rootURL = url
+        self.configuration = configuration
+
+        do {
+            try FileManager.default.createDirectory(at: rootURL, withIntermediateDirectories: true, attributes: nil)
```

Comment:
Does this need to be part of the initializer?

We can't guarantee that the OS won't remove this directory on us as part of its usual disk pressure cleanup.

WDYT about ensuring that the directory tree exists when creating a new cache entry?

Reply:
> WDYT about ensuring that the directory tree exists when creating a new cache entry?
Yes, the initial implementation already takes care of it.